nav2_smac_planner/src/a_star.cpp

```diff
@@ -192,35 +191,106 @@ template<>
 void AStarAlgorithm<Node2D>::setGoal(
   const float & mx,
   const float & my,
-  const unsigned int & dim_3)
+  const unsigned int & dim_3,
+  const GoalHeadingMode & /*goal_heading_mode*/,
+  const int & /*coarse_search_resolution*/)
 {
   if (dim_3 != 0) {
     throw std::runtime_error("Node type Node2D cannot be given non-zero goal dim 3.");
   }
 
-  _goal = addToGraph(
+  NodePtr goal = addToGraph(
     Node2D::getIndex(
       static_cast<unsigned int>(mx),
       static_cast<unsigned int>(my),
       getSizeX()));
-  _goal_coordinates = Node2D::Coordinates(mx, my);
+
+  goal->setPose(Node2D::Coordinates(mx, my));
+  GoalStateVector goals_state;
+  goals_state.push_back({goal, true});
+  _goal_manager.setGoalStates(
+    goals_state);
+  _coarse_search_resolution = 1;
 }
 
 template<typename NodeT>
 void AStarAlgorithm<NodeT>::setGoal(
   const float & mx,
   const float & my,
-  const unsigned int & dim_3)
+  const unsigned int & dim_3,
+  const GoalHeadingMode & goal_heading_mode,
+  const int & coarse_search_resolution)
 {
-  _goal = addToGraph(
-    NodeT::getIndex(
-      static_cast<unsigned int>(mx),
-      static_cast<unsigned int>(my),
-      dim_3));
+  // Default to minimal resolution unless overridden for ALL_DIRECTION
+  _coarse_search_resolution = 1;
+
+  unsigned int num_bins = NodeT::motion_table.num_angle_quantization;
+  // all state would be marked as valid
+  GoalStateVector goals_state;
+
+  // set goal based on heading mode
+  switch (goal_heading_mode) {
+    case GoalHeadingMode::DEFAULT: {
+        // add a single goal node with single heading
+        auto goal = addToGraph(NodeT::getIndex(mx, my, dim_3));
+        goal->setPose(typename NodeT::Coordinates(
+          static_cast<int>(mx), static_cast<int>(my), static_cast<int>(dim_3)));
+        goals_state.push_back({goal, true});
+        break;
+      }
 
-  typename NodeT::Coordinates goal_coords(mx, my, dim_3);
+    case GoalHeadingMode::BIDIRECTIONAL: {
+      // Add two goals, one for each direction
```

Comment:
The comments and code aren't aligned (same as later switch cases)

Reply:
done.